src/huggingface_hub/hf_api.py

```diff
@@ -7607,6 +7611,14 @@ def create_inference_endpoint(
                 Secret values to inject in the container environment.
             type ([`InferenceEndpointType]`, *optional*):
                 The type of the Inference Endpoint, which can be `"protected"` (default), `"public"` or `"private"`.
+            domain (`str`, *optional*):
+                The custom domain for the Inference Endpoint (e.g. `"my-new-domain.cool-website.woof"`).
+            path (`str`, *optional*):
+                The custom path for the Inference Endpoint, should start with a `/` (e.g. `"/models/google-bert/bert-base-uncased"`).
+            cache_http_responses (`bool`, *optional*):
```

Comment:
(link this [private thread](https://huggingface.slack.com/archives/C016D661PAN/p1744979768406779?thread_ts=1742997838.592509&cid=C016D661PAN) to avoid forgetting about it - if this attribute is HF-only, let's remove it)